src/buffer.js

```diff
@@ -0,0 +1,127 @@
+const {contextRecordKey, deferred} = require('./utils');
+
+function queryBuffer(config, emitter, targetStore) {
+  const buffers = [];
+
+  let numCoalesced = 0;
+  let numCached = 0;
+  let numMisses = 0;
+
+  class RequestBuffer {
+    constructor(key, params) {
+      this.uid = Math.random().toString(36);
+      this.state = 0;
+      this.ids = [];
+      this.contextKey = key;
+      this.params = params;
+      this.contexts = [];
+      this.handle = deferred();
+      this.timer = null;
+    }
+
+    tick() {
+      const sizeLimit = config.batch?.limit || 1;
+
+      if (this.ids.length >= sizeLimit) {
+        this.run('limit');
+        return this;
+      }
+      if (this.timer === null) {
+        this.timer = setTimeout(this.run.bind(this, 'timeout'), config.batch?.delay || 0);
```

Comment:
Same as above

Reply:
The bind here is probably more expensive, but that's the price to pay to use Classes in JS -.-

Reply:
I guess if this was of paramount importance we could set these parameters once in the constructor and just call `this.config.whathever`!

Reply:
I'll still approve :)